### ISSUE TITLE
Floats return as strings in 3.24.4

### ISSUE THREAD

**Issue by yktv4** [sequelize/sequelize#6763]
## What you are doing?
I do a simple select of a float number and see what different sequelize versions return
```js
sequelize.query('SELECT 3.22 AS shouldBeFloat')
  .spread(result => console.log(result, typeof result[0].shouldbefloat));
```

## What do you expect to happen?
I would like floats to be returned as floats and not strings OR if it's not possible anymore, I'd like to have consistent library behavior which would not break my application by going a couple minor versions up.

## What is actually happening?
floats are returned as strings in sequelize version `3.24.4` though they were returned as numbers at least in sequelize `3.19.3`. Here's what snippet above outputs for different sequelize versions:

*sequelize 3.19.3:* `[ { shouldbefloat: '3.22' } ] 'string'`
*sequelize 3.24.4:* `[ { shouldbefloat: 3.22 } ] 'number'`

__Dialect:__ postgres
__Database version:__ 9.5.3
__Sequelize version:__ 3.24.4 vs 3.19.3

**Comment by yktv4** [comment]
I've found out that this is due to this PR
https://github.com/sequelize/sequelize/pull/5860

I see the reasons underneath it, but I believe returning the whole another type of data is a breaking change and should not be applied in a minor version upgrade `3.23.0 -> 3.23.1`

**Comment by janmeier** [comment]
I see where you are coming from, but parsing decimals into javascript floats was a bug that was introduced in 3.14 https://github.com/sequelize/sequelize/issues/4893, and then fixed again in 3.23.1
